### ISSUE TITLE
Delete instance method removes formatting / Whitespace in <pre> tags slows down writing speed 

### ISSUE THREAD

**Issue by HJEC** [alexmacarthur/typeit#204]
<em>Hey there!</em>

First of all, I'd just like to say how much <strong>I love typeit </strong>, I've been using the react component version and I think it's the best thing since sliced bread.

I'm not sure if this should be separated into two different issues, but I'm really new to all this so here we go. 

<h4> The project: </h4>
So I'm working on an intro sequence for my "for-fun" portfolio project stuff, in which I'm trying to emulate an old PC boot sequence, using typeit to print out some jargon text, and then an ASCII art version of the logo for the intro page of the site.

<h4> The problems:</h4>
1. The instance delete.() method seems to collapse the entire span and remove any formatting, line-breaks and such, before carrying on. I've enclosed a gif of the problem here: 

![delete](https://user-images.githubusercontent.com/52827871/79008261-40f20800-7b5d-11ea-8746-1fc91b3d9664.gif)

I tried to simulate a loading sequence by writing out "...", deleting it and again etc. After the delete method, it continues writing out with linebreaks as usual. I tried a mix of using **.type("\<br/>")** or the **.break()** methods to see if was effected differently, but it didn't seem to make a difference.

I'm not sure if this is expected behaviour of the delete method, but It's unfortunate for my use case. 
here is a snippet of the instance where I used delete: 
```
.type("LOCALE time is: ", { speed: 10, delay: 300 })
.break()
.type("Fetching location:  ")
 .type(" ...", {
    speed: 100,
    delay: 300
})
.delete(4, {speed: 1, delay: 300})
.type(" ...", {
   speed: 100,
   delay: 300
})
.delete(4, {speed: 1, delay: 300})
.type("<br/> <br />")
.type(fillText(location.country + "<br/>"), {
       speed: 50
       })
```

2. Whitespace inside of \<pre> tags seems to exponentially slow down the write speed for each line of the typeit component. Here is another gif of the problem:

![intro_slowest](https://user-images.githubusercontent.com/52827871/79009658-5288df00-7b60-11ea-85f6-e38c38b0414e.gif)

During the second phase of the intro screen, I used a second TypeIt component to start printing out an ASCII version of a logo. I wanted to achieve an effect of the logo being printed really quickly, like you would imagine from a retro movie. 
Perhaps I'm using typeit for a grossly innapropriate task, but like I mentioned, I'm new to this and I really like the flexibility of typeit so far. 

I tried to achieve this by importing a template literal that starts and ends with \<pre> tags, and some ascii art I generated online, and then feeding that to the .type() method, here is a screenshot for reference: 

![2020-04-10 (2)](https://user-images.githubusercontent.com/52827871/79010006-3cc7e980-7b61-11ea-8287-be792a9bfb6b.png)

It appears that the whitespace appearing in between each set of strings seems to be slowing the write speed down more and more until it gets to a crawl.
This is what the component looks like, where "asciiLogo" is just the name of the imported module.

```
<TypeIt
     id="ascii"
     options={{
               speed: 1,
               lifeLike: false,
               cursor: false,
               html: true
      }}
      getBeforeInit={instance => {
                instance.type(asciiLogo);
                return instance;
      }}
      ></TypeIt>
```

I wondered if there was an option that I was missing that would prevent this whitespace related slowdow, like "nextStringDelay" perhaps?

Once again, thanks so much for this library, and sorry if this issue is a bit of a mess. Forgive me, I'm new! 
Cheers :)

**Comment by alexmacarthur** [comment]
Thanks for reporting this! All the detail is especially appreciated -- will look into it and see what I can do. Probably aiming for later this weekend, maybe early next week.

**Comment by alexmacarthur** [comment]
Good news -- the first issue you reported is fixed and shouldn't be a problem anymore. Try out v7.0.2 of TypeIt itself or v0.1.1 of the React component to verify this. 

As for the second issue -- this'll be a larger problem to solve. What I think is happening is that as so many characters are typed, the amount of work it takes to append more and more characters becomes much more of a performance challenge. It sort of has to do with the method I'm using to append new characters, which will be difficult to move away from, since that particular method offers unique benefits I need elsewhere. 

For the time being, I've made a separate issue to track work & discovery on this. Would definitely love this to not be a problem for use cases like this, even though they're probably relatively rare. 

https://github.com/alexmacarthur/typeit/issues/205

**Comment by HJEC** [comment]
@alexmacarthur Wow, thank you for replying so quickly! That's really helpful.
Yeah I assumed that the amount of characters inside the template literal being interpreted were going to have an effect on the method. I noticed how much white-space there was in the string and I figured it was taking the time to process that before appending the next character in the span which caused the slowdown. 
I managed to create a quick-and-dirty fix by breaking up the ASCII art logo into two seperate typeit components, this removed about 75% of the white-space and improved the write speed. Then I used a setTimeout() to pause the "chat" section from writing until it should appear on screen. It's not perfect, but then again things rarely are!

Once again, I'd just like to say thanks for this awesome, awesome utility!
